crates/factor-outbound-http/src/wasi.rs

```diff
@@ -297,14 +319,25 @@ async fn send_request_handler(
         }
     } else {
         let tcp_stream = TokioIo::new(tcp_stream);
-        let (sender, conn) = timeout(
-            connect_timeout,
-            // TODO: we should plumb the builder through the http context, and use it here
-            hyper::client::conn::http1::handshake(tcp_stream),
-        )
-        .await
-        .map_err(|_| ErrorCode::ConnectionTimeout)?
-        .map_err(hyper_request_error)?;
+        let (sender, conn) = if is_http2 {
```

Comment:
Is it feasible to deduplicate this now rather large and complicated block of code with the one around line 289-307?